account_payment_term_cutoff_day/models/account_payment_term_line.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2025 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import fields, models
+from odoo.tools import date_utils
+
+
+class AccountPaymentTermLine(models.Model):
+    _inherit = "account.payment.term.line"
+
+    cutoff_day = fields.Integer(
+        default=31,
+        help="Specify the cutoff day of the month for adjusting invoice due dates."
+        "For example, if you set this field to 20, any invoice dated on the 21st or "
+        "later will have its due date moved to the following month."
+        "Setting 0 is treated the same as having no cutoff day, which in practice is"
+        "the same as setting 31.",
+    )
+
+    def _get_due_date(self, date_ref):
```

Comment:
IMO, the core design was better in 16.0 and it now lacks the flexibility for adjustments (i.e., we cannot specify a payment day when a cutoff day is set). Could we redesign this module so that it depends less on the core design? For example, we could add a boolean field `has_cutoff_day` to `account.payment`. If the field is selected, the logic could fall back on what we implemented in 16.0. We can make necessary adjustments to the views.

Adjusting the code to handle each `delay_type` could still be an option, but I doubt that the effort would be worth it.